File: day3/pages/base_page.py
class BasePage:
    def __init__(self, url):
        self.url        = url
        self.is_open    = False
        self.page_title = ""

    def navigate(self):
        self.is_open = True
        print(f"🌐 Navigating to: {self.url}")

    def get_title(self):    
        print(f"📄 Page title: {self.page_title}")


    def close(self):
        self.is_open = False
        print(f"❌ Closing: {self.url}")

File: week3/pages/base_page.ts
//pages/dase_page.ts
//Base page shared by ALL page objects
// Manisha Wadhwa
// Week 3 Day 3 — Page Object Model (POM) in Playwright


import { Page } from '@playwright/test';

export class BasePage {
//page = the browser window
protected page: Page;

constructor(page: Page) {
    this.page = page;
}


//Navigate to a URL
async navigate(url: string) {
    await this.page.goto(url);
    console.log(`✅ Navigated to ${url}`);
}


//get current page title
async getTitle(): Promise<string> { 
       return await this.page.title();   
}


//Get current page URL
async getURL(): Promise<string> {
    return this.page.url();

}

//wait for page to load
async waitForPageLoad() {
    await this.page.waitForLoadState('networkidle');
}

}

